### PR TITLE
Fix deal.II (and SPARTA) run_with_generator + modern CMake minimum (#39)

### LINKED ISSUE #39: Version conflict while using Oasis  and Deal.ii
Hi,  I have tried to install the OASiS on ARM MAC  with Claude Code and installed the Deal.ii from a dmg package.  Here are the issues that Claude Code found (and solved).
deal.II compilation issues and fixes
1. OASiS's run_with_generator tool is broken for deal.II jobs. Its input-file detection only globs for *.4C.yaml, *.yaml, input.*, solve.py, MainKratos.py — never .cpp. So it always reported "Generator did not produce an input file" even though the generator script correctly wrote main.cpp + CMakeLists.txt to disk. Fix: skip that tool for the actual build; write the files, then run cmake/make/the binary directly via Bash in the job's output directory.

2. OASiS's auto-generated CMakeLists.txt is stale. It hardcodes cmake_minimum_required(VERSION 3.1), which your installed CMake 4.4.0 rejects outright (removed support for <3.5). Fix: write my own CMakeLists.txt with a modern minimum version.

3. deal.II's own bundled compiler has a baked-in SDK conflict. deal.II lives at /Applications/deal.II.app (spack-built). Its mpicxx wrapper carries a hardcoded -isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include from one of its dependencies (likely Trilinos/Kokkos), which collides with your active (beta) Xcode SDK's libc++ headers — every compile failed inside deal.II's own exceptions.h with abs/isinf conflicts in <complex>/<cmath>. Reordering include paths via CMake didn't help (CMake silently drops entries that match its "implicit" system dirs), and stripping the path from the imported dealii::dealii_debug target didn't help either (it comes from a deeper transitive dependency). Fix: a small compiler-wrapper shell script that calls the real mpicxx but strips that one bad -isystem pair of arguments before forwarding — set via -DCMAKE_CXX_COMPILER=<wrapper>.

### PR DESCRIPTION
**Pull request by alhermann**
## Summary

Fixes #39. deal.II simulations through `run_with_generator` were completely broken. This PR fixes that, a related CMake-version issue, and a second backend (SPARTA) that had the identical detection gap.

## The bugs

1. **`run_with_generator` never detected deal.II output.** The input-file glob `["*.4C.yaml","*.yaml","input.*","solve.py","MainKratos.py"]` (duplicated in `consolidated.py` and `simulation.py`) matches no `.cpp`, so deal.II's `main.cpp` was invisible and a *successful* generator was reported as *"Generator did not produce an input file"*. SPARTA's `in.sparta` had the same problem (`input.*` does not match `in.sparta`).
2. **Generated `CMakeLists.txt` hardcoded `cmake_minimum_required(VERSION 3.1)`.** CMake >= 4 removed support for `< 3.5`, so every deal.II build failed on recent CMake (the reporter's 4.4.0 rejected it outright).

## The fix

- New backend-aware `find_generated_input()` helper in `core/backend.py`, keyed on `InputFormat`, with a legacy-pattern fallback and a last-resort scan so no valid generator output is ever dropped. Both call sites use it.
- `cmake_minimum_required` `3.1 -> 3.13.4` (deal.II's documented minimum) in the three deal.II spots.

## Verification (Linux, deal.II installed)

- A real deal.II program compiled with `3.13.4`, linked deal.II, ran, and returned the correct result.
- The exact issue #39 flow: generator writes `main.cpp` + a stale `3.1` CMakeLists -> detected -> recompiled at `3.13.4` -> ran -> correct result.
- SPARTA `in.sparta` detection now works end-to-end (launches and runs real DSMC steps).
- deal.II passes the full 10-benchmark cross-solver e2e suite; 30 deal.II/generator/backend unit tests are green.
- New regression test `tests/test_generated_input_detection.py` (11 tests): all six input formats plus edge cases (decoy files, `input_format` as method vs attribute, generator-script exclusion, empty dir -> None).

## Notes (out of scope, pre-existing)

- The deal.II.app bundled `mpicxx` SDK conflict (issue #39 item 3) is a third-party-binary / environment problem, not OASiS. The code already honours `CC`/`CXX`, so the reporter's compiler wrapper plugs in via `-DCMAKE_CXX_COMPILER`.
- SPARTA's example decks reference a species file (`ar.species`) the generator does not ship — a separate template gap, worth its own issue.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/tools/consolidated.py [hunk 1]
```diff
@@ -1478,6 +1474,7 @@ async def run_with_generator(solver: str, generator_script: str,
                 "status": "failed", "phase": "generator",
                 "error": "Generator did not produce an input file",
                 "work_dir": str(work_dir),
+                "files": [f.name for f in work_dir.iterdir()],
```
[the remark this answers is not in the document]
Comment (alhermann) [reply src/tools/consolidated.py]: Good catch — addressed in 6624e70: the list is now capped to 50 sorted filenames, so the response stays bounded and deterministic. Applied to the equivalent path in simulation.py too.
